### DISCUSSION TITLE
TShock 5.1.3 for Terraria 1.4.4.9 (Thankyou!! <3)

### DISCUSSION THREAD

**Discussion by hakusaro** [Pryaxis/TShock#2843] [Announcements]
Welcome to TShock for Terraria 5.1.3 with support for Terraria 1.4.4.9. This is, of course, also on nuget.

**Before installing, make a complete backup of your existing server and all files and folders, including `tshock.sqlite` and the `tshock` folder. We are not responsible for failed installations. Please make sure you backup your data. Severe data loss or gain may occur.**

## Notable changes in this release
* Added support for Terraria 1.4.4.9 via OTAPI 3.1.20. (@SignatureBeef)

## Upgrading

**Before installing, make a complete backup of your existing server and all files and folders, including `tshock.sqlite` and the `tshock` folder. We are not responsible for failed installations. Please make sure you backup your data. Severe data loss or gain may occur.**

To upgrade TShock, replace all of the files in the location of your server installation with all of the files from the release zip we provide. Overwrite all files. No data will be lost.

<hr /><em>This discussion was created from the release <a href='https://github.com/Pryaxis/TShock/releases/tag/v5.1.3'>TShock 5.1.3 for Terraria 1.4.4.9 (Thankyou!! <3)</a>.</em>

**Comment by lyavkube** [comment]
Thanks for update!

**Comment by gaster181** [comment]
why i cant start TShock?

**Reply by gaster181** [reply]
something appears but immediately disappears

**Reply by drunderscore** [reply]
You are likely missing the [.NET 6.0 runtime](https://dotnet.microsoft.com/en-us/download/dotnet/6.0) -- try installing for your OS/arch and re-trying.

Unfortunately, Microsoft made it very ominous with no obvious user-facing errors when missing the runtime for a console application. This will be somewhat mitigated by #2840, which will allow most platforms and architectures to run an installer to easily get a .NET runtime specifically for TShock, simply by executing `TShock.Installer`.

**Reply by gaster181** [reply]
Thanks

**Comment by 6epkytob** [comment]


**Reply by sgkoishi** [reply]
Try a fresh installation. There are some legacy files in your directory which are not compatible with latest version.

**Comment by 6epkytob** [comment]
Help please. I'm using a virtual machine Linux Ubuntu.
I created a user. Installed the Tshock kernel version 5.1.3 in it. Then I started running using the mono command.
I got an error: Cannot open assembly 'TShock.Server.exe': File does not contain a valid CIL image.

**Reply by Arthri** [reply]
TShock 5 uses .NET 6 instead of .NET Framework/Mono. The download instructions(for Linux distros) is available here https://learn.microsoft.com/en-us/dotnet/core/install/linux-scripted-manual#manual-install

**Reply by 6epkytob** [reply]
I installed NET6, but the error remains. 

root@vd2563:/# ./dotnet-install.sh --version latest
dotnet-install: Note that the intended use of this script is for Continuous Integration (CI) scenarios, where:
dotnet-install: - The SDK needs to be installed without user interaction and without admin rights.
dotnet-install: - The SDK installation doesn't need to persist across multiple CI runs.
dotnet-install: To set up a development environment or to run apps, use installers rather than this script. Visit https://dotnet.microsoft.com/download to get the installer.
dotnet-install: .NET Core SDK with version '6.0.403' is already installed.
root@vd2563:/# ./dotnet-install.sh --version latest --runtime aspnetcore
dotnet-install: Note that the intended use of this script is for Continuous Integration (CI) scenarios, where:
dotnet-install: - The SDK needs to be installed without user interaction and without admin rights.
dotnet-install: - The SDK installation doesn't need to persist across multiple CI runs.
dotnet-install: To set up a development environment or to run apps, use installers rather than this script. Visit https://dotnet.microsoft.com/download to get the installer.
dotnet-install: ASP.NET Core Runtime with version '6.0.11' is already installed.
root@vd2563:/# cd home/terraria
root@vd2563:/home/terraria# mono TShock.Server.exe
Cannot open assembly 'TShock.Server.exe': File does not contain a valid CIL image.

**Reply by Arthri** [reply]
Do not use Mono. Run TShock.Server directly instead. Also make sure to set the correct environment variables as described in the linked page

**Reply by john-rothbard** [reply]
Im getting this same issue, running on a Raspberry pi 4. What do you mean by Run TShock.Server directly? I am new to linux

**Reply by sgkoishi** [reply]
> Im getting this same issue, running on a Raspberry pi 4. What do you mean by Run TShock.Server directly? I am new to linux

![image](https://user-images.githubusercontent.com/9637711/227760165-926f08a4-536b-4033-bf8e-6ddfcf55b529.png)
<sup>Answered by the [new Bing](https://www.bing.com/chat)</sup>

**Reply by john-rothbard** [reply]
that results in "bash: ./TShock.Server: No such file or directory" :/

**Reply by sgkoishi** [reply]
> that results in "bash: ./TShock.Server: No such file or directory" :/

![image](https://user-images.githubusercontent.com/9637711/227760501-a17c0668-54f3-4a4b-8191-61e3d7d379a3.png)

**Reply by Arthri** [reply]
What have you done so far? You need to download the zip file first then unzip and untar it. Assuming your current working directory contains the zip file, Run the command `unzip TShock-5.1.3-for-Terraria-1.4.4.9-linux-arm64-Release.zip` followed by `tar -xf TShock-Beta-linux-arm64-Release.tar`

**Reply by Arthri** [reply]
You also need to have ARM64 or x64 PI. **Do not** use x32 or ARM32 PI

**Reply by ali3n7** [reply]
Hi, I have my Pi4 running and have the ZIP for ARM64 extracted. I'm in the folder for TShock and try to start TShock.Server but get always the message "Command not found". Sorry for the novice question. And thanks for all hints. Alex
![2023-04-03 16_43_45-sys-server-raspi4 (SYS-Server-Raspi4) – VNC Viewer](https://user-images.githubusercontent.com/129767791/229544494-cfa5c9cb-fe1e-4470-88d9-7fa17bbddbab.png)

**Reply by Arthri** [reply]
Try `./TShock.Server` with `./` in front

**Reply by ali3n7** [reply]
cannot execute binary. Error in binary format

**Reply by Arthri** [reply]
Do you have the .NET 6 Runtime installed? If not, you need it, make sure to install the correct architecture though(in this case, ARM64)

**Reply by ali3n7** [reply]
thanks first. Now I'm confused. Had an old tut for installing TShock on a Pi, installed Mono and before installing TShock I recognized, that the newer version is no longer using Mono. Does that mean that it need .NET 6 Runtime for Pi? Did not installed this right now (seenoevil)

**Reply by Arthri** [reply]
Yes, TShock 5 does not use mono anymore, it uses .NET 6

**Reply by ali3n7** [reply]
same error. Looks like I have only 32 Bit. grrrr
how can I deinstall the 64Bbit version? Not that firm handling Linux

![image](https://user-images.githubusercontent.com/129767791/229552527-ba79caae-9355-4f82-a4b8-fd8e4e8bd975.png)

**Reply by Arthri** [reply]
Sorry, I've never worked with Pi

**Reply by ali3n7** [reply]
sorry again me. Just recognized, that I didn't install anything, so just cleaned the folder and downloaded and upacked the ARM version.
no I get the following error. But I installed the .NET 6
Any other hints?

![image](https://user-images.githubusercontent.com/129767791/229557057-8a5e4f97-d16c-4de8-8a93-5001ef397898.png)

![image](https://user-images.githubusercontent.com/129767791/229559609-d70ad17d-7c30-4df0-a386-2019963356bf.png)

**Reply by ali3n7** [reply]
Variables were not set properly. Followed the instructions in the link to the MS page. Thanks for any help

**Comment by set-night** [comment]
![image](https://user-images.githubusercontent.com/85575488/209708387-949e92a1-5112-4819-aa6a-4c2c90832822.png)
Tshok cant find .net
![image](https://user-images.githubusercontent.com/85575488/209708505-78dd8de2-2c5a-460a-bdd0-433d48a01910.png)

**Reply by sgkoishi** [reply]
The [dev version](https://github.com/Pryaxis/TShock/actions/workflows/ci-otapi3.yml?query=event%3Apush+branch%3Ageneral-devel) comes with an installer and it will download and setup things for you.

**Comment by quickuserlll** [comment]
Rasberian os support? I did everything dotnet itp. and it still doesn't work.

**Reply by sgkoishi** [reply]
The [dev version](https://github.com/Pryaxis/TShock/actions/workflows/ci-otapi3.yml?query=event%3Apush+branch%3Ageneral-devel) comes with an installer and it will download and setup things for you.

**Reply by quickuserlll** [reply]
so what I'm supposed to do with this information

**Reply by sgkoishi** [reply]
https://github.com/Pryaxis/TShock/discussions/2881#discussioncomment-4446200
Raspberry is ARM or ARM64 depends on your model, so `TShock-Beta-linux-arm-Release` or `TShock-Beta-linux-arm64-Release`, just download either and try.

**Reply by quickuserlll** [reply]
ye it is an ARM64 but I still have no idea what I have to do I don't know how or what you want me to download the #2871 leads me to Linux arm64 actions and I don't know what do I do there/next

**Reply by quickuserlll** [reply]
![Zrzut ekranu z 2022-12-27 23-31-54](https://user-images.githubusercontent.com/117534629/209728680-7ca7a73a-0c55-4196-adc6-13df2fc34025.png)

**Reply by sgkoishi** [reply]
> Click the first one with CI(OTAPI3) (for now it is [this one](https://github.com/Pryaxis/TShock/actions/runs/3683343469)), scroll down and find your version.

The link points to the `Summary` page, you don't need to click `build (linux-arm64)`. Click back to the `Summary`, scroll to the bottom.

**Reply by quickuserlll** [reply]
ye

**Reply by quickuserlll** [reply]
o FUCK I AM AN IDIOT

**Reply by quickuserlll** [reply]
I AM SO sorry

**Reply by quickuserlll** [reply]
thx for the help

**Reply by sgkoishi** [reply]
> The logs for this run have expired and are no longer available.

@tab960562 Just confirmed all links under this comment are still valid. No idea how did you find that expired link.

**Comment by kaeru4** [comment]
Something pops up and disappears, I have windows 11 (.NET 6.0 is installed too)
What can I do to open it without it disappearing? What can be the problem?

**Reply by sgkoishi** [reply]
Shift+Right click on empty space, choose `Open in Terminal` or `Open in cmd` or Powershell, drag the TShock.Server.exe to the new window and press enter.
Or, try [dev version](https://github.com/Pryaxis/TShock/discussions/2843#discussioncomment-4504794) mentioned in earlier comments.

**Comment by Gatt00** [comment]
nothing works, I run TShock.Server.exe, the console appears for a millisecond and immediately disappears, I have .NET installed, I don’t understand what’s wrong.

**Reply by sgkoishi** [reply]
https://github.com/Pryaxis/TShock/discussions/2881#discussioncomment-4445111

**Reply by Gatt00** [reply]
it didn't help me at all

**Reply by sgkoishi** [reply]
http://www.catb.org/~esr/faqs/smart-questions.html#idm573, if you don't have any info to add

**Reply by Gatt00** [reply]
Why do I need a site with rules? I need to run tshock and not read this nonsense

**Reply by Gatt00** [reply]
can you just explain to me directly why the tishok does not start?

**Reply by sgkoishi** [reply]
The first link clearly said:
![image](https://user-images.githubusercontent.com/9637711/216454834-7f838063-7655-4802-8af5-1930ac96a4da.png)

**Reply by Gatt00** [reply]
please

**Reply by Gatt00** [reply]
Unable to load developer version for windows

**Reply by sgkoishi** [reply]
If you can't take a screenshot because it exits, the first link also said:
![image](https://user-images.githubusercontent.com/9637711/216455692-dbee83d5-becf-4793-bc92-22d4dd32b768.png)

Replace the `TShock.Server` with `TShock.Installer` for the dev version.

**Reply by Gatt00** [reply]
let's say I downloaded this file → https://github.com/Pryaxis/TShock/commit/6e589ffaeed99fa1b1559a5aa18c375d04961fb3#annotation_7023032712
what should I do with it next?

**Reply by sgkoishi** [reply]
[How to download (this is also copied from the first link):](https://github.com/Pryaxis/TShock/discussions/2881#discussioncomment-4446200)
> From the link you can find many workflow runs. Click the first one with `CI(OTAPI3)` (for now it is [this one](https://github.com/Pryaxis/TShock/actions/runs/3683343469)), scroll down and find your version (you are using Windows, so you want the last one `TShock-Beta-win-x64-Release`). There is a `TShock.Installer.exe` inside.

**Reply by Gatt00** [reply]
thank you everything is working

**Reply by Gatt00** [reply]
Can you tell me what are the colors for the /group color groups, and how to set them correctly?

**Reply by Gatt00** [reply]
and how to open it for the network?

**Reply by Gatt00** [reply]
Or does it only work on a local connection?

**Reply by Gatt00** [reply]
I wanted to play with a friend

**Reply by sgkoishi** [reply]
Colours are [rgb colour](https://www.google.com/search?q=rgb+color), for example, pure red is `255,0,0` and white is `255,255,255`.
For networking, you need a public IP address. You can try and see if [your IP](https://www.google.com/search?q=my%20ip%20address) is the right one; it should work in most cases.

**Reply by Gatt00** [reply]
IP address is public, how to open a server for the network?

**Reply by sgkoishi** [reply]
It is open if nothing blocks it. If you are running on your PC, allow the firewall when prompted.

**Reply by Gatt00** [reply]
Thanks, can I ask another question, why did I choose 16 players when creating the server, but when I enter the server, 1/8 players write?

**Reply by Gatt00** [reply]
i wanted 16 players :(

**Reply by sgkoishi** [reply]
[MaxSlots](https://github.com/Pryaxis/TShock/blob/general-devel/docs/config-file-descriptions.md#maxslots)

**Reply by Gatt00** [reply]
it needs to be changed in config.json?

**Reply by hakusaro** [reply]
Yes. Can you move this to discord and stop emailing over 100 people every five minutes? This isn’t an IM service.

**Reply by Gatt00** [reply]
I'm not on your discord

**Comment by yeahbuddy601** [comment]
I wanted to update my old tshock 4.3.26 on my raspberry but it is not booting. Before I used to do mono TerrariaServer.exe and now also mono Tshock.Server.exe does not work. 
Thanks in advance!

**Reply by sgkoishi** [reply]
https://github.com/Pryaxis/TShock/discussions/2843#discussioncomment-4349572

**Reply by yeahbuddy601** [reply]
Quick Question: if I delete everything and make a completely new server with all steps and the latest tshock and mono, this CIL problem won’t occur?

**Reply by sgkoishi** [reply]
https://github.com/Pryaxis/TShock/discussions/2843#discussioncomment-4351214

**Reply by yeahbuddy601** [reply]
So not install Mono software at all? But the Windows tshock version is not the right one for raspberry pi then? (If I would do a completly new server) Or should I use linux ARM tshock.zip?

**Reply by sgkoishi** [reply]
TShock 5 does not use mono, so you don't have to install it. Linux ARM64 for rpi usually.

**Comment by Danefron** [comment]
What is ``error CS1503: Argument 1: cannot convert from 'int' to 'Terraria.DataStructures.IEntitySource' ``

**Comment by Steveve007** [comment]
I'm running Mac OS and after I ran the tshock.server I couldn't find the tshock folder.
Any help?

**Reply by sgkoishi** [reply]
Download the latest [v5.2.0](https://github.com/Pryaxis/TShock/releases/) and run the installer.

**Reply by Steveve007** [reply]
Do I run it with the terminal?

**Reply by Steveve007** [reply]
because whenever I open the tshock.installer with the terminal no tshock folder pops up, but when I use the archive utility it creates a new file with .cpgz then I opened it again with the terminal, and no tshock folder.

**Reply by sgkoishi** [reply]
Yes, with terminal. If it is not behaving as expected, what message does it show, and can you take a screenshot?

**Reply by Steveve007** [reply]
pls scroll down to the end of screen shots.
<img width="1440" alt="Screenshot 2023-07-19 at 3 08 26 PM" src="https://github.com/Pryaxis/TShock/assets/139934098/b1afb44b-72f4-4c62-be82-e61a4d94406a">

**Reply by Steveve007** [reply]
<img width="1440" alt="Screenshot 2023-07-19 at 3 08 02 PM" src="https://github.com/Pryaxis/TShock/assets/139934098/5d7c9ac7-37f3-4630-8a90-9893d003a0db">

**Reply by Steveve007** [reply]
<img width="1440" alt="Screenshot 2023-07-19 at 3 07 23 PM" src="https://github.com/Pryaxis/TShock/assets/139934098/682ca8e4-5c3a-463b-96f9-dee432ebce3b">

**Reply by Steveve007** [reply]
i created a tshock folder and if it did anything but nothing

**Reply by Steveve007** [reply]
<img width="1440" alt="Screenshot 2023-07-19 at 3 10 43 PM" src="https://github.com/Pryaxis/TShock/assets/139934098/fd1f803e-f1a2-4586-8828-5633939c5df1">

**Reply by sgkoishi** [reply]
Which one did you launch? It seems that you have a `terraria hosting server` and a `TShock-Beta-osx-x64-Release` on your desktop - are you using different ones?

**Reply by Steveve007** [reply]
Yeah I'm testing 2 different ones to see which one creates a folder but both didn't sadly. I mainly use the terraria hosting server cause that's where my plugins are.

**Reply by sgkoishi** [reply]
Can you try `exit` the TShock and run `pwd` and `ls` in the terminal?

**Reply by Steveve007** [reply]
I don't know what pwd and ls is

**Reply by Steveve007** [reply]
<img width="1440" alt="Screenshot 2023-07-19 at 3 33 17 PM" src="https://github.com/Pryaxis/TShock/assets/139934098/4aea1a9c-affa-4133-8edd-ebfa6412dcab">

**Reply by Steveve007** [reply]
here's ls

**Reply by sgkoishi** [reply]
So here it is - your TShock is running in `Macintosh HD > Users > stevenguyen`, not `Macintosh HD > Users > stevenguyen > Desktop` or `Macintosh HD > Users > stevenguyen > Desktop > terraria hosting servers`.

You can `cd ~/Desktop/something` in terminal before you run TShock.

**Reply by Steveve007** [reply]
yea im trying to watch a video of cd pwd and ls so i understand a bit now

**Reply by Steveve007** [reply]
whenever I try to change to the subdirectory terraria hosting servers  it says too many arguments or do I just leave it like this /Users/stevenguyen/desktop

**Reply by Steveve007** [reply]
<img width="1440" alt="Screenshot 2023-07-19 at 3 49 45 PM" src="https://github.com/Pryaxis/TShock/assets/139934098/4fc4c6c2-800e-42af-87eb-763023abdc3c">
not sure if its working rn

**Reply by sgkoishi** [reply]
Use `'single quote'` or `"double quote"` if you have space between, or press `Tab` to let it auto-complete for you.
You may also try to open terminal from your TShock folder (I don't know how to do that in Mac, but it should be easy)

**Reply by Steveve007** [reply]
thx for all the help I got all the files in the tshock folder and configs now tysm and it's really appreciated.

**Comment by aarvndhNG** [comment]
Is it possible to run tshock on Google colab

**Reply by ghost** [reply]
does it cost money?

**Comment by aarvndhNG** [comment]
> reply

Idk i think no
